### PR TITLE
Fixed Guards not removing armor and weapons when fired.

### LINKED ISSUE #3462: Bug:  Citizen still wearing armor after changing career from guard to something else.
### Minecolonies version
CaveColonies

Version: 

This bug has been around for a long time.  I apologize if it is already listed as an issue as I could not figure out how to effectively search the GitHub.  No time to read 180+ items. =p

### Expected behavior
Citizens do not wear armor and do not carry weapons or tools when in civilian clothing or after switching careers.

### Actual behavior
A citizen with all level 1 attributes was hired as a guard and given a bow and leather armor.  They were fired as a guard and hired as a student.  They continued to wear the leather armor as a student.  Later, they were employed as a cook and persisted to wear the armor.

### Steps to reproduce the produce

1. Hire a guard (ranger).
2. Give them leather armor and a bow.
3. Fire them and rehire in a different occupation.
4. Observe their attire.

### PR DESCRIPTION
**Pull request by Elrol**
Closes #3462 

# Changes proposed in this pull request:
- Removes the equiped items on a Guard when they are fired

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java [hunk 1]
```diff
@@ -427,8 +427,12 @@ public void removeCitizen(final CitizenData citizen)
             if (optCitizen.isPresent())
             {
                 optCitizen.get().removeAllHealthModifiers();
-
-                optCitizen.get().getEntityAttribute(SharedMonsterAttributes.ARMOR).setBaseValue(SharedMonsterAttributes.ARMOR.getDefaultValue());
+                optCitizen.get().setItemStackToSlot(EntityEquipmentSlot.CHEST, ItemStackUtils.EMPTY);
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:430]: Does this move the item to his normal inventory?
Or does this delete it?

**Review by marchermans — COMMENTED** [review]
Some questions....

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java (hunk 1 above)
Comment (Asherslab) [reply src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:430]: i can answer this. the armour is equipped while keeping the item in his inventory, same with tools and etc. as such there's technically two copies of the same items (one in armor slot and one in inventory) . but yes i had him test this and it keeps the items in his inventory, while removing them from his "armor slots"

Comment (Elrol) [reply src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:430]: The items never leave his inventory. if you remove the armor from his inventory he will remove it, and all this does is force him to remove the armor. I tested it on my end and it works like expected.

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/3558/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Comment by Asherslab** [comment]
LGTM


[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/3558/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Pull request merged**